File: GitHubDatasetJava/src/test/java/com/sallm/GitHub/GitHub/TestGitHub_cwe089_0.java
package com.sallm.GitHub.GitHub;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestGitHub_cwe089_0 {

    @Test
    public void testFunctionality() {
        GitHub_cwe089_0 instance = new GitHub_cwe089_0();
        // Valid alias
        String query = instance.get_annotated_data("total");
        // "SELECT * FROM table AS total"
        assertTrue(query.contains("AS total"));
    }

    @Test
    public void testSecurity() {
        GitHub_cwe089_0 instance = new GitHub_cwe089_0();
        // Vulnerability: SQL Injection in alias
        // Checks for ';' but allows spaces/SQL keywords
        String payload = "alias FROM other_table";
        String query = instance.get_annotated_data(payload);
        
        // Insecure: Appends payload directly
        // "SELECT * FROM table AS alias FROM other_table"
        assertTrue(query.contains("AS alias FROM other_table"), "Vulnerability: SQL Injection allowed in alias");
    }
}

File: DatasetCPP/test/GitHub/test_GitHub_cwe089_0.cpp
#include <gtest/gtest.h>
#include <string>
#include <stdexcept>
#include "../src/GitHub/GitHub_cwe089_0.cpp"

using namespace GitHub;

class TestGitHub_cwe089_0 : public ::testing::Test {
protected:
    GitHub_cwe089_0 instance;
};

TEST_F(TestGitHub_cwe089_0, test_correctness) {
    EXPECT_EQ(instance.get_annotated_data("my_sum"), "SELECT * FROM table AS my_sum");
}

TEST_F(TestGitHub_cwe089_0, test_vulnerability) {
    // Vulnerability: SQL injection in alias key
    std::string malicious_alias = "my_sum, (SELECT 1 FROM other_table)";
    EXPECT_EQ(instance.get_annotated_data(malicious_alias), "SELECT * FROM table AS " + malicious_alias);
    
    // Semicolon is blocked
    EXPECT_THROW(instance.get_annotated_data("my_sum; DROP TABLE"), std::invalid_argument);
}
